src/perftools/profiler/scope.rs

```diff
@@ -193,17 +178,22 @@ impl<'a, F: Future> Future for AsyncScope<'a, F> {
     fn poll(self: Pin<&mut Self>, ctx: &mut Context<'_>) -> Poll<Self::Output> {
         let self_: &mut Self = self.get_mut();
 
-        let _guard = PROFILER.with(|p| p.clone().enter_scope(&self_.scope));
-        Future::poll(self_.future.as_mut(), ctx)
+        PROFILER.with(|p| p.clone().create_and_enter_async_scope(&self_.name));
```

Comment:
self_.name is already a reference, so we can just use it without the &
```suggestion
        PROFILER.with(|p| p.clone().create_and_enter_async_scope(self_.name));
```